client/my-sites/people/team-list/test/team.jsx

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+
+import React from 'react';
+import { shallow } from 'enzyme';
+import { expect } from 'chai';
+import sinon from 'sinon';
+
+/**
+ * Internal dependencies
+ */
+import useFakeDom from 'test/helpers/use-fake-dom';
+import useMockery from 'test/helpers/use-mockery';
+
+describe( 'Team', function() {
+	let Card;
+	let PeopleListSectionHeader;
+	let InfiniteList;
+	let Team;
+	let configMock;
+	let deterministicStringify;
+
+	useFakeDom();
+	useMockery( mockery => {
+		configMock = sinon.stub();
+		configMock.isEnabled = sinon.stub();
+		mockery.registerMock( 'config', configMock );
+
+		deterministicStringify = sinon.stub();
+		mockery.registerMock( 'lib/deterministic-stringify', deterministicStringify );
+		mockery.registerMock( 'lib/sites-list', () => null );
+	} );
+
+	before( function() {
+		Card = require( 'components/card' );
+		PeopleListSectionHeader = require( 'my-sites/people/people-list-section-header' ).default;
+		InfiniteList = require( 'components/infinite-list' );
+
+		Team = require( '../team' ).Team;
+	} );
+
+	it( 'renders infinite list when site is specified and any users have been loaded', function() {
+		deterministicStringify
+			.withArgs( { option1: 'option1value', option2: 'option2value' } )
+			.returns( 'list key' );
+
+		const wrapper = shallow(
+			<Team
+				fetchInitialized={ true }
+				fetchingUsers={ false }
+				fetchOptions={ { option1: 'option1value', option2: 'option2value', number: 10, offset: 2 } }
+				site={ 'site1' }
+				users={ [ 'user1' ] }
+				excludedUsers={ [] }
+			/>
+		);
+
+		expect( wrapper.find( Card ).find( InfiniteList ).key() ).to.equal( 'list key' );
+	} );
+
+	it( 'renders translated team header text by default', function() {
+		const translate = sinon.stub()
+			.withArgs( 'Team' )
+			.returns( 'team label' );
+
+		const wrapper = shallow(
+			<Team
+				fetchInitialized={ true }
+				fetchingUsers={ false }
+				fetchOptions={ {} }
+				users={ [] }
+				excludedUsers={ [] }
+				translate={ translate }
+			/>
+		);
+
+		expect( wrapper.find( PeopleListSectionHeader ).prop( 'label' ) )
+			.to.equal( 'team label' );
+	} );
+
+	it( 'renders translated header text for search when there are search results', function() {
+		const translate = sinon.stub()
+			.returns( 'translated search label' );
+
+		const wrapper = shallow(
+			<Team
+				fetchInitialized={ true }
+				fetchingUsers={ false }
+				fetchOptions={ {} }
+				search={ 'abc' }
+				site={ 'site1' }
+				totalUsers={ 1 }
+				users={ [ 'user1' ] }
+				excludedUsers={ [] }
+				translate={ translate }
+			/>
+		);
+
+		expect( wrapper.find( PeopleListSectionHeader ).prop( 'label' ) )
+			.to.equal( 'translated search label' );
+	} );
```

Comment:
Well there's a certain balance to this. I'm always a fan of unit tests when it helps shape interfaces, and helps describe original intent to future readers. Component tests have a bit less value since they do require more maintenance and may require some white box testing approaches. Normally I only write these when it's a very commonly used component, or it has regressed in prod and we don't want it to happen again. If something has regressed a lot even with component tests, I might add an integration smoke test, the most fragile of our test types, to our e2e project to make sure we know when it breaks in prod.

I'm sure other folks have thoughts on this. Maybe cc @gziolo, @aduth or @blowery